tests/certification/bindings/azure/storagequeues/storagequeue_test.go

```diff
@@ -0,0 +1,527 @@
+				secretstores_loader.New("local.env", func() secretstores.SecretStore {
+					return secretstore_env.NewEnvSecretStore(log)
+				}),
+			))).
+		Step("send and wait", testDecode).
+		Run()
+}
+
+func TestAzureStorageQueueRetriesOnError(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+	messages := watcher.NewUnordered()
+
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+	appPort := ports[2]
+
+	testRetry := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		// Declare the expected data.
+		msgs := make([]string, numOfMessages)
+		for i := 0; i < numOfMessages; i++ {
+			msgs[i] = fmt.Sprintf("Message %03d", i)
+		}
+
+		messages.ExpectStrings(msgs...)
+
+		metadata := make(map[string]string)
+
+		// Send events that the application above will observe.
+		ctx.Log("Invoking binding!")
+		for _, msg := range msgs {
+			ctx.Logf("Sending: %q", msg)
+
+			req := &daprClient.InvokeBindingRequest{Name: "retry-binding", Operation: "create", Data: []byte(msg), Metadata: metadata}
+			err := client.InvokeOutputBinding(ctx, req)
+			require.NoError(ctx, err, "error publishing message")
+		}
+
+		// Assertion on the data.
+		messages.Assert(ctx, time.Minute)
+
+		return nil
+	}
+	// Application logic that tracks messages from a topic.
+	retryApplication := func(ctx flow.Context, s common.Service) (err error) {
+		// Simulate periodic errors.
+		sim := simulate.PeriodicError(ctx, 10)
+
+		// Setup the input binding endpoint.
+		err = multierr.Combine(err,
+			s.AddBindingInvocationHandler("retry-binding", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+				if err := sim(); err != nil {
+					ctx.Logf("Failing message: %s", string(in.Data))
+					return nil, err
+				}
+				messages.Observe(string(in.Data))
+				ctx.Logf("Got message: %s", string(in.Data))
+				return []byte("{}"), nil
+			}))
+
+		return err
+	}
+
+	flow.New(t, "servicebusqueue retry certification").
+		// Run the application logic above.
+		Step(app.Run("retryApp", fmt.Sprintf(":%d", appPort), retryApplication)).
+		Step(sidecar.Run("retrySidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/retry"),
+			runtime.WithOutputBindings(
+				binding_loader.NewOutput("azure.storagequeues", func() bindings.OutputBinding {
+					return binding_asq.NewAzureStorageQueues(log)
+				}),
+			),
+			runtime.WithInputBindings(
+				binding_loader.NewInput("azure.storagequeues", func() bindings.InputBinding {
+					return binding_asq.NewAzureStorageQueues(log)
+				}),
+			),
+			runtime.WithSecretStores(
+				secretstores_loader.New("local.env", func() secretstores.SecretStore {
+					return secretstore_env.NewEnvSecretStore(log)
+				}),
+			))).
```

Comment:
Could you please send some data before interrupting the network to show everything was working prior to the interruption?

Reply:
It's not that important really - so never mind :)